test/test-lib/common.ts

```diff
@@ -36,3 +37,35 @@ export async function waitFor({
 
 	throw new TimedOutError();
 }
+
+export const itExpectsError = (
+	title: string,
+	fn: Mocha.AsyncFunc,
+	isExpectedError?: string | RegExp | ((err: Error) => boolean),
```

Comment:
```suggestion
	expectedError?: string | RegExp | ((err: Error) => boolean),
```
`isExpectedError` makes sense for the function, but not for the others. I'd call is `expectedError` and update the function case to return the expected error string to make it better match the other options.

Reply:
If I get this right you suggest to do `(err: Error) => string`.
How will then the returned error message be used by the helper?
That parameter is used by the code as a filter on whether the correct error was thrown, and the string & regex alternatives are there just for convenience.
If anything, I would expect us to actually convert the string & regex to a matcher function (in the helper's body), so that we can have a single v check.
```
if (!isExpectedError(err)) {
	throw err;
}
```

Reply:
I was thinking a function that returned a string that you then compared to the actual error message string in the helper. But converting the string and regex to a matcher function like you mentioned would be fine as well I think, as long as each option follows the same general usage pattern.